More flake8 errors and warnings

Running on current master (7885b0cf1ddde4cc0f471a34d4f7e6ffc76f7a0b):

```bash
$ flake8 --version
3.7.0 (mccabe: 0.6.1, pycodestyle: 2.5.0, pyflakes: 2.1.0) CPython 3.6.8 on Linux
$ flake8 | wc -l
590
$ flake8 src/pymor/ | wc -l
77
$ flake8 src/pymortests/ | wc -l
293
$ flake8 src/pymordemos/ | wc -l
110
$ flake8 | grep E741 | wc -l
67
$ flake8 src/pymor/ | grep E741 | wc -l
47
$ flake8 src/pymor/ | grep -v E741
src/pymor/reductors/residual.py:93:121: E501 line too long (121 > 120 characters)
src/pymor/reductors/residual.py:244:121: E501 line too long (124 > 120 characters)
src/pymor/operators/mpi.py:9:1: F401 'pymor.vectorarrays.numpy.NumpyVectorSpace' imported but unused
src/pymor/operators/mpi.py:60:48: E711 comparison to None should be 'if cond is None:'
src/pymor/operators/numpy.py:165:12: F632 use ==/!= to compare str, bytes, and int literals
src/pymor/gui/qt.py:199:14: E401 multiple imports on one line
src/pymor/gui/qt.py:359:63: E127 continuation line over-indented for visual indent
src/pymor/core/pickle.py:25:1: F401 'sys' imported but unused
src/pymor/core/pickle.py:77:1: E303 too many blank lines (4)
src/pymor/core/cache.py:295:5: E731 do not assign a lambda expression, use a def
src/pymor/core/cache.py:408:13: E117 over-indented
src/pymor/core/cache.py:419:37: E117 over-indented (comment)
src/pymor/core/config.py:17:1: E302 expected 2 blank lines, found 1
src/pymor/tools/mpi.py:123:13: E722 do not use bare 'except'
src/pymor/tools/mpi.py:185:13: E722 do not use bare 'except'
src/pymor/tools/mpi.py:301:33: E127 continuation line over-indented for visual indent
src/pymor/tools/quadratures.py:236:13: E117 over-indented
src/pymor/algorithms/basic.py:56:9: E731 do not assign a lambda expression, use a def
src/pymor/algorithms/timestepping.py:160:15: E127 continuation line over-indented for visual indent
src/pymor/algorithms/timestepping.py:161:15: E127 continuation line over-indented for visual indent
src/pymor/algorithms/image.py:278:51: F821 undefined name 'image'
src/pymor/algorithms/gram_schmidt.py:75:57: F821 undefined name 'old_norm'
src/pymor/algorithms/gram_schmidt.py:90:17: F841 local variable 'old_norm' is assigned to but never used
src/pymor/algorithms/gram_schmidt.py:168:59: F821 undefined name 'old_norm'
src/pymor/algorithms/error.py:192:23: E128 continuation line under-indented for visual indent
src/pymor/parameters/functionals.py:114:9: E731 do not assign a lambda expression, use a def
src/pymor/parameters/spaces.py:98:9: E731 do not assign a lambda expression, use a def
src/pymor/discretizers/fv.py:144:19: F821 undefined name 'nonlinear_advection_upwind_operator'
src/pymor/discretizers/cg.py:287:87: E713 test for membership should be 'not in'
src/pymor/analyticalproblems/text.py:47:57: E117 over-indented (comment)
```

So, there are still plenty of code style issues. Most of them are not even in `src/pymor`. Of those in `src/pymor` most are about using `l` and `I` for variable names (E741), but there are still many other types.

@pymor/pymor-devs How do we decide which issues to ignore (by updating setup.cfg or just "mentally") and which to fix?

--- a/setup.cfg
+++ b/setup.cfg
@@ -2,11 +2,13 @@
 test = pytest
 [pep8]
 max-line-length = 120
-ignore = E221,E226,E241,E242, W0105, W503, N803, N806
+ignore = E221,E226,E241,E242, E731, E741, W0105, W503, N803, N806
 # E221 multiple spaces before operator
 # E226 missing whitespace around arithmetic operator  [ignored by default]
 # E241 multiple spaces after ':'                      [ignored by default]
 # E242 tab after `,'                                  [ignored by default]
+# E731 do not assign a lambda expression, use a def
+# E741 do not use variables named 'l', 'O', or 'I'
 # W0105 String statement has no effect (we use triple qoted strings as documentation in some files)
 # W503 line break before binary operator
 # N803 argument name should be lowercase (we use single capital letters everywhere for vectorarrays)
@@ -15,7 +17,7 @@ ignore = E221,E226,E241,E242, W0105, W503, N803, N806
 
 [flake8]
 max-line-length = 120
-ignore = E221,E226,E241,E242, W0105, W503, N803, N806
+ignore = E221,E226,E241,E242, E731, E741, W0105, W503, N803, N806
 # The following exclude avoids wrong warnings for unused imports
 exclude = __init__.py
 

--- a/setup.cfg
+++ b/setup.cfg
@@ -3,6 +3,7 @@ test = pytest
 
 [pycodestyle]
 max-line-length = 120
+max-doc-length = 100
 ignore = E221,E226,E241,E242, E731, E741, W0105, W503, N803, N806
 # E221 multiple spaces before operator
 # E226 missing whitespace around arithmetic operator  [ignored by default]
@@ -17,6 +18,7 @@ ignore = E221,E226,E241,E242, E731, E741, W0105, W503, N803, N806
 
 [flake8]
 max-line-length = 120
+max-doc-length = 100
 ignore = E221,E226,E241,E242, E731, E741, W0105, W503, N803, N806
 # The following exclude avoids wrong warnings for unused imports
 exclude = __init__.py

--- a/setup.cfg
+++ b/setup.cfg
@@ -1,6 +1,7 @@
 [aliases]
 test = pytest
-[pep8]
+
+[pycodestyle]
 max-line-length = 120
 ignore = E221,E226,E241,E242, E731, E741, W0105, W503, N803, N806
 # E221 multiple spaces before operator
@@ -13,7 +14,6 @@ ignore = E221,E226,E241,E242, E731, E741, W0105, W503, N803, N806
 # W503 line break before binary operator
 # N803 argument name should be lowercase (we use single capital letters everywhere for vectorarrays)
 # N806 same for variables in function
-
 
 [flake8]
 max-line-length = 120

--- a/src/pymor/discretizers/fv.py
+++ b/src/pymor/discretizers/fv.py
@@ -140,10 +140,6 @@ def discretize_stationary_fv(analytical_problem, diameter=None, domain_discretiz
         if num_flux == 'lax_friedrichs':
             L += [nonlinear_advection_lax_friedrichs_operator(grid, boundary_info, p.nonlinear_advection,
                                                               dirichlet_data=p.dirichlet_data, lxf_lambda=lxf_lambda)]
-        elif num_flux == 'upwind':
-            L += [nonlinear_advection_upwind_operator(grid, boundary_info, p.nonlinear_advection,
-                                                      p.nonlinear_advection_derivative,
-                                                      dirichlet_data=p.dirichlet_data)]
         elif num_flux == 'engquist_osher':
             L += [nonlinear_advection_engquist_osher_operator(grid, boundary_info, p.nonlinear_advection,
                                                               p.nonlinear_advection_derivative,

